### PR TITLE
Feature/updated constants

### PR DESCRIPTION
**Pull request by jlconlin**
In this Pull Request, I've updated the physical constants to something a bit more modern. Of course, this changes virtually all of the reference tapes. Exactly what to do with this is a matter of discussion.

### CODE REVIEW

**Comment by jlconlin** [comment]
Note: @kahlerac did most of the work on this and gave me the upn file before he retired. I incorporated the changes into git and had to make a few minor changes to get it to compile.

**Comment by whaeck** [comment]
So I looked at the changes in the output files. Can you rerun the tests using a tolerance of 1e-5? That'll allow us to quantify the overall impact of changing the values of the physical constants.

**Comment by jlconlin** [comment]
I tested with 1E-5 relative tolerance. This didn't change the test results. 

When the comparison is done, the first thing that is done is to count the number of lines in the the generated file and the reference file. If the number of lines are not the same, then no other checking is done. In fact, additional checking can't be done since we do everything line-by-line.

**Comment by jlconlin** [comment]
Now, perhaps we should change the way that we compare references files. There is an argument to be made there.

**Comment by whaeck** [comment]
Well, we do test non regression, so if the number of lines is different there is definitely something going on. In those cases a by hand verification is warranted.

I assume that in this case, no files actually get additional lines?

**Comment by jlconlin** [comment]
No, there are plenty that get additional lines.

**Comment by jlconlin** [comment]
I've updated the constants even more. I double-checked the values put in previously. I used the constants in the ENDF manual and from NIST (when not in the ENDF manual).

File: src/samm.f90 [hunk 1]
```diff
@@ -1300,27 +1295,27 @@ subroutine ppdefs(ier)
...
          lpent(i,ier)=1
       else if (mt(i,ier).eq.106) then  ! he3
-         ema(i,ier)=he3/neutron
+         ema(i,ier)=amassh/amassn
          spina(i,ier)=0.5e0_kr
          kza(i,ier)=2
          lpent(i,ier)=1
       else if (mt(i,ier).eq.107) then  ! alpha
-         ema(i,ier)=alpha/neutron
+         ema(i,ier)=amassa/amassn
```
Comment (whaeck) [inline src/samm.f90]: See above: these values return a couple of times. Maybe introduce the following parameters:
pnratio=amassp/amassn
dnratio=amassd/amassn
etc.

File: src/thermr.f90 [hunk 2]
```diff
@@ -2300,7 +2300,7 @@ subroutine calcem(temp,itemp,iold,inew,ne,nex)
       write(nsyso,'(/''            mu            theta      dsigma/dmu'')')
       do i=1,nmu
          write(nsyso,'(i5,1x,f15.8,1x,f12.4,1x,1p,e14.7)') i,uj(i),&
-               acos(uj(i))*180.0/3.14159265359,sj(i)/2.0
+           acos(uj(i))*180/pi,sj(i)/2
```
Comment (whaeck) [inline src/thermr.f90]: This was the one that caused a lot of issues in the thermr outputs? Try acos(uj(i))*180.0/pi,sj(i)/2.0 instead to see if it continues.

**Comment by whaeck** [comment]
By the way, do not forget to update the version number and the date.

File: src/samm.f90 (hunk 1 above)
Comment (jlconlin) [reply src/samm.f90]: Yeah, but where else are they used? Are they used outside of `samm.f90`? If not, then it shouldn't be part of `phys.f90`. I also don't want to make this a big refactoring of of the entire codebase. We have to draw the line somewhere.

Comment (whaeck) [reply src/samm.f90]: groupr too.

File: src/leapr.f90 [hunk 3]
```diff
@@ -1959,14 +1957,19 @@ subroutine coldh(itemp,temp)
...
-   if (law.gt.3) sampi=sampid
+   if (law.gt.3) then
+     amassm=6.69E-24_kr
+     ! amassm=2*(amassd+amasse)*amu*ev/(clight*clight)
+     sampc=sampcd
+     bp=hbar/2*sqrt(2/ded/ev/dmass)/angst
+     sampi=sampid
+   else
+     amassm=3.3464e-24_kr
+     ! amassm=2*(amassp+amasse)*amu*ev/(clight*clight)
```
[the remark this answers is not in the document]
Comment (jlconlin) [reply src/leapr.f90]: When I change this to use the formula, I get a different value for `amassm`, different by 12 orders of magnitude! So I don't think that formula is correct. I chose to leave it as is until we know what is going on.

**Pull request merged**